lib/features/email/presentation/controller/single_email_controller.dart

```diff
@@ -1718,40 +1720,78 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
         break;
     }
   }
+
+  String? getLanguageForCalendarEvent(
+    Locale locale,
+    CalendarEventCapability calendarEvent
+  ) {
+    final supportedLanguages = calendarEvent.replySupportedLanguage;
+    if (supportedLanguages == null) return null;
+
+    final currentLanguage = locale.languageCode;
+    if (supportedLanguages.contains(currentLanguage)) {
+      return currentLanguage;
+    } else if (supportedLanguages.contains('en')) {
+      return 'en';
+    } else {
+      return supportedLanguages.firstOrNull;
+    }
+  }
   
   void _acceptCalendarEventAction() {
     if (_acceptCalendarEventInteractor == null
       || _displayingEventBlobId == null
-      || mailboxDashBoardController.accountId.value == null) {
-        consumeState(Stream.value(Left(CalendarEventAcceptFailure())));
+      || mailboxDashBoardController.accountId.value == null
+      || mailboxDashBoardController.sessionCurrent?.getCapabilityProperties<CalendarEventCapability>(
+        mailboxDashBoardController.accountId.value!,
+        CapabilityIdentifier.jamesCalendarEvent) == null
```

Comment:
IMO, should be create method in session_extension for validate it.

Reply:
Done